### PR TITLE
Add gamepad support

### PR DESCRIPTION
**Pull request by moeabm**
add gamepad support - https://developer.mozilla.org/en-US/docs/Web/API/Gamepad_API

### CODE REVIEW

**Comment by robsdedude** [comment]
Great idea and thanks for the support! I'm currently rather busy, but will hopefully be able to test and merge this PR within the next days.

**Comment by moeabm** [comment]
Its a bit of a different use case for me but I see other uses for it. Basically, Im building a retropie for a facility that has cc tv modulators. The retropie will feed for example CH14 UHF. People at their desk can tune their TV into the retropie video feed and plug in a controller to their desktop PC to control it. The physical gamepad experience is a big win.

However, I could see how others would just prefer the gamepad to be physically connected to their mobile device.

File: app/virtual_gamepad_hub.js [hunk 1]
```diff
@@ -44,7 +44,8 @@ Virtual gamepad hub class
         this.gamepads[padId] = new gamepad();
         return this.gamepads[padId].connect(function() {
           return callback(padId);
-        }, function(err) {
+        }, (err)=> {
+          this.gamepads[padId] = void 0;
```
Comment (robsdedude) [inline app/virtual_gamepad_hub.js]: Changing things in the server's JS code won't do much as it's written in coffee script and the JS code is generated from it. I will take care of an equivalent change in the coffee script.

**Comment by robsdedude** [comment]
Thanks again for the work. Please note that it would have been easier if you had based your work on and filed the PR on the brach `develop`.

**Pull request merged**